sqlglot/dialects/tsql.py

```diff
@@ -708,6 +715,33 @@ def parse_range():
 
             return partition
 
+        def _parse_declare(self) -> exp.Declare | exp.Command:
+            expressions = self._try_parse(partial(self._parse_csv, self._parse_declareitem))
+            if not expressions:
```

Comment:
The parsing may succeed, so we'd get a non-falsey `expressions` here, but there may be leftover tokens that we haven't consumed and so we'd produce an incomplete AST node. This addition ensures we will still fallback into parsing a command for these cases.
```suggestion
            if not expressions or self._curr:
```